src/dc_msg.rs

```diff
@@ -831,98 +832,74 @@ pub unsafe fn dc_msg_get_summarytext(
         return dc_strdup(0 as *const libc::c_char);
     }
 
-    let msgtext_c = (*msg)
-        .text
-        .as_ref()
-        .map(|s| CString::yolo(String::as_str(s)));
-    let msgtext_ptr = msgtext_c.map_or(ptr::null(), |s| s.as_ptr());
-
     dc_msg_get_summarytext_by_raw(
         (*msg).type_0,
-        msgtext_ptr,
+        (*msg).text.as_ref().unwrap(),
         &mut (*msg).param,
         approx_characters,
         (*msg).context,
     )
+    .strdup()
 }
 
-/* the returned value must be free()'d */
+/// get a summary text
 #[allow(non_snake_case)]
```

Comment:
I think you should aim to remove this as well if you're cleaning up this function.